crates/sui-core/src/authority/execution_time_estimator.rs

```diff
@@ -425,6 +448,52 @@ impl ExecutionTimeObserver {
         self.share_observations(to_share);
     }
 
+    fn generate_test_timings(
+        &self,
+        tx: &ProgrammableTransaction,
+        timings: &[ExecutionTiming],
+    ) -> (Vec<ExecutionTiming>, Duration) {
+        let generated_timings: Vec<_> = tx
+            .commands
+            .iter()
+            .zip(timings.iter())
+            .map(|(command, timing)| {
+                let key = ExecutionTimeObservationKey::from_command(command);
+                let duration = self.get_test_duration(Some(&key));
+                if timing.is_abort() {
+                    ExecutionTiming::Abort(duration)
+                } else {
+                    ExecutionTiming::Success(duration)
+                }
+            })
+            .collect();
+
+        let total_duration = generated_timings
+            .iter()
+            .map(|t| t.duration())
+            .sum::<Duration>()
+            + self.get_test_duration(None);
```

Comment:
I think the last term here will always be a constant value for the duration of a test? Instead we should use `thread_rng().gen_range(...)` to add random noise. Also, the noise should probably vary from 0 to some smallish value (maybe 50?)